### PR TITLE
Add from and to for linker in state step

### PR DESCRIPTION
**Pull request by lixingwang**
**What kind of change does this PR introduce?** (check one with "x")
```
[*] Bugfix
[] Feature
[] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[] Other... Please describe:
```

**Fixes**: #

**What is the current behavior?**
1. Debugger state mode does not enable snapshot which should be enabled.
2. There is no link from and to from linker in state step, it much better/easy to understand if we add from and to
**What is the new behavior?**
1. Enable snapshot save for debugger model
2. Add from and to for linker in state  step

### CODE REVIEW

File: instance/instance_chg.go [hunk 1]
```diff
@@ -250,6 +250,8 @@ func (sct *SimpleChangeTracker) TaskRemoved(subflowId int, taskId string) {
 func (sct *SimpleChangeTracker) LinkAdded(linkInst *LinkInst) {
```
Comment (fm-tibco) [inline instance/instance_chg.go:250]: Isn't this redundant information that can be retrieved from the link definition?

File: state/status.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func RecordSteps(stateRecordingMode RecordingMode) bool {
 // RecordSnapshot check to see if enbale snapshot recording
 func RecordSnapshot(stateRecordingMode RecordingMode) bool {
 	switch stateRecordingMode {
-	case RecordingModeSnapshot, RecordingModeFull:
+	case RecordingModeSnapshot, RecordingModeFull, RecordingModeDebugger:
```
Comment (fm-tibco) [inline state/status.go:49]: You can technically generate a snapshot from the step recording... so you might not need this.

File: instance/instance_chg.go (hunk 1 above)
Comment (lixingwang) [reply instance/instance_chg.go:250]: @fm-tibco This is about link in state step data.   Today it just has `change` and `status`,   we don't know the link from which task to which task.  that's why I added here.   The steps save in state server which we don't have a definition.

Comment (lixingwang) [reply instance/instance_chg.go:250]: We enabled it for debugger mode only.

**Pull request merged**